### PR TITLE
[JBIDE-20599] Allow the handler to set the connection instead of recent

### PR DESCRIPTION
**Pull request by jcantrill**
@fbricon review?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImagePage.java [hunk 1]
```diff
@@ -231,11 +231,18 @@ private void createProjectControl(Composite parent, DataBindingContext dbc) {
...
+		final OpenShiftExplorerLabelProvider labelProvider = new OpenShiftExplorerLabelProvider();
 		cmboProject.setContentProvider(new ObservableListContentProvider());
-		cmboProject.setLabelProvider(new OpenShiftExplorerLabelProvider());
+		cmboProject.setLabelProvider(labelProvider);
 		cmboProject.setInput(
 				BeanProperties.list(IDeployImagePageModel.PROPERTY_PROJECTS).observe(model));
-		
+		cmboProject.setSorter(new ViewerSorter() {
+			@Override
+			public int compare(Viewer viewer, Object e1, Object e2) {
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImagePage.java]: shouldn't `default` always be on top?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java [hunk 2]
```diff
@@ -81,7 +81,6 @@
 	private boolean originatedFromDockerExplorer;
 	
 	public DeployImageWizardModel() {
-		setConnection(ConnectionsRegistrySingleton.getInstance().getRecentConnection(Connection.class));
```
Comment (fbricon) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java]: why is this removed?

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImagePage.java (hunk 1 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImagePage.java]: It should, but maybe we should consider removing it all together.  I'm not sure how the current ording works in the explorer window.  I assume it returns in order of what was returned by the call.  I dont know whay it wouldnt do the same here.

File: plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java (hunk 2 above)
Comment (jcantrill) [inline plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java]: This is what was causing the problem.  It returns the one recently accessed but we want the one you selected.  So I defered it to being set in the handler. Hmm... maybe we need to explicitly set 'recent' when calling from the docker explorer side?

**Comment by fbricon** [comment]
Applied in master and 4.3.x
